osu.Game/Screens/Edit/Screens/Setup/Components/SwitchButton.cs

```diff
@@ -0,0 +1,142 @@
+        public Color4 DisabledColour
+        {
+            get => disabledColour;
+            set
+            {
+                if (!Current.Value)
+                    AccentColour = value;
+                disabledColour = value;
+            }
+        }
+
+        private Color4 accentColour;
+        public Color4 AccentColour
+        {
+            get => accentColour;
+            set
+            {
+                accentColour = value;
+                circularContainer.BorderColour = value;
+            }
+        }
+
+        public SwitchButton()
+        {
+            Container switchCircle;
+            Size = new Vector2(size_x, size_y);
+
+            InternalChild = circularContainer = new CircularContainer
+            {
+                RelativeSizeAxes = Axes.Both,
+                BorderColour = Color4.White,
+                BorderThickness = border_thickness,
+                Masking = true,
+                Children = new Drawable[]
+                {
+                    fill = new Box
+                    {
+                        RelativeSizeAxes = Axes.Both,
+                        AlwaysPresent = true,
+                        Alpha = 0
+                    },
+                    new Container
+                    {
+                        RelativeSizeAxes = Axes.Both,
+                        Padding = new MarginPadding(border_thickness + switch_padding),
+                        Child = switchContainer = new Container
+                        {
+                            RelativeSizeAxes = Axes.Both,
+                            Child = switchCircle = new CircularContainer
+                            {
+                                RelativeSizeAxes = Axes.Both,
+                                FillMode = FillMode.Fit,
+                                Masking = true,
+                                Child = new Box { RelativeSizeAxes = Axes.Both }
+                            }
+                        }
+                    }
+                }
+            };
+
+            Current.ValueChanged += newValue =>
+            {
+                if (newValue)
+                    switchCircle.MoveToX(switchContainer.DrawWidth - switchCircle.DrawWidth, 200, Easing.OutQuint);
+                else
+                    switchCircle.MoveToX(0, 200, Easing.OutQuint);
+                this.FadeAccent((newValue ? enabledColour : disabledColour).Lighten(IsHovered ? 0.3f : 0), 500, Easing.OutQuint);
+                fill.FadeTo(newValue ? 1 : 0, 500, Easing.OutQuint);
+            };
+        }
+
+        [BackgroundDependencyLoader]
+        private void load(OsuColour colours)
+        {
+            EnabledColour = colours.BlueDark;
+            DisabledColour = colours.Gray3;
+            switchContainer.Colour = enabledColour;
+            fill.Colour = disabledColour;
+        }
+
+        protected override bool OnClick(InputState state)
+        {
+            Current.Value = !Current.Value;
```

Comment:
`Current.Toggle()`

Reply:
A cast to `BindableBool` would have to be performed beforehand, should we go with this?

Reply:
`public BindableBool Current { get; } = new BindableBool();`

Reply:
It requires a `Bindable<Bool>` if we want it to use the interface `IHasCurrentValue<bool>`, otherwise  removing the interface constraint would make it work properly.